src/v/cloud_topics/level_zero/stm/ctp_stm.h

```diff
@@ -98,6 +98,10 @@ class ctp_stm final : public raft::persisted_stm<> {
     ss::rwlock _lock;
     /// Current in-memory state of the STM
     ctp_stm_state _state;
+
+    // The last observed epoch to be applied to the state machine. This value is
+    // used to check for violations of monotonicity in epoch order.
+    cluster_epoch _last_seen_epoch{};
```

Comment:
Should this be snapshotted and restored? I get that this is only there just in case, but it seems we could have stronger validation if this took into account the previous state